extra/CMakeLists.txt

```diff
@@ -33,6 +33,16 @@ set(BUILD_TYPE_C_FLAGS "${CMAKE_C_FLAGS_${_BUILD_TYPE}}")
 set(BUILD_TYPE_CXX_FLAGS "${CMAKE_CXX_FLAGS_${_BUILD_TYPE}}")
 message("CXX Compile Flags from BUILD_TYPE: ${BUILD_TYPE_CXX_FLAGS}")
 
+############################################################
+# Disable new-dtags with newer linkers
+############################################################
+
+if (UNIX AND NOT APPLE)
```

Comment:
If you want to avoid this workaround I think we could live with RUNPATH by setting it also in the shared libraries (apparently it allows that). Shall I experiment with that?

Reply:
without that --disable-dtags option, the RUNPATH is populated correctly but still get library not found error. Didnt understand that quite well.

Reply:
It might be related to the fact that they are used a bit differently by the dynamic loader: See [Bug #1253638 “dynamic linker does not use DT_RUNPATH for transitive dependencies](https://bugs.launchpad.net/ubuntu/+source/eglibc/+bug/1253638)
If I understood well, RUNPATH is only valid for direct dependencies, but sub-dependencies can also define RUNPATH which gets also checked. I kind of agree with the new behavior, but true, breaks things